### PR TITLE
Using `@testWith` annotation may generate `PHP Warning: Uninitialized string offset 0`

### PR DESCRIPTION
**Pull request by staabm**
Closes https://github.com/sebastianbergmann/phpunit/issues/5795

### CODE REVIEW

File: tests/end-to-end/regression/5795/Issue5795Test.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+use PHPUnit\Framework\TestCase;
+
+final class Issue5795Test extends TestCase
+{
+    /**
+     * @testWith		 [1]
+     * 			 [2]
+     * 			 [3]
+     *
```
Comment (staabm) [inline tests/end-to-end/regression/5795/Issue5795Test.php:20]: for the test to fail this line needs to contain a space after the `*`.

my local phpstorm automatically strips this space and therefore I was not able to reproduce

**Comment by staabm** [comment]
btw: I never seen this feature used in the wild. I am not sure when to use it, if one case use a regular DataProvider.

writing data for tests in a phpdoc comment, which will be parsed and mangled by a lot of other tools sounds like a footgun.

**Comment by sebastianbergmann** [comment]
> I never seen this feature used in the wild

Me neither.

> I am not sure when to use it, if one can use a regular DataProvider instead

This was proposed in #1151 and implemented in #1728. The argument was that "[data providers are] overcomplicated and require to[o] much code for most tasks".

I should have said "No!" at the time, but did not :-(

> writing data for tests in a phpdoc comment, which will be parsed and mangled by a lot of other tools sounds like a footgun

At least this footgun will be gone once support for annotations in comments is removed.

**Comment by staabm** [comment]
> I should have said "No!" at the time, but did not :-(

Without trying we wouldn't know. So it was a small failed experiment.. just look forward to :-)

**Pull request merged**